estate/models/estate_property.py

```diff
@@ -127,3 +128,55 @@ def _onchange_garden(self):
         else:
             self.garden_area = 0
             self.garden_orientation = False
+
+    def _check_availability(self, error_msg):
+        """ Errors en error if the property is not available. Raises an
+        UserError when the property is not available for sale, that is, if it
+        is sold or canceled. Receives a single record in `self` and an error
+        message.
+        """
+        self.ensure_one()
+        if self.state in ["sold", "canceled"]:
+            message = _("%s property: %s", self.state.capitalize(), error_msg)
+            raise UserError(message)
+
+    def _clear_offers(self):
+        """ Change state of offers to "refused". Set the state of all offers
+        to "refused". Receives a single record in `self`.
+        """
+        self.ensure_one()
```

Comment:
Please. Delete line

Reply:
`self.ensure_one()`?

Reply:
Yes, , for the following reason:

If in the following line, what I follow is a `for` on the `self`, it means that you are supporting multiple records.

Reply:
The next line does a `for` over `offer_ids` (which is a recordset of offers), but it assumes a single record (`self`).